engine/src/vs_logging.h

```diff
@@ -48,41 +57,63 @@ enum vega_log_level {
     fatal
 };
 
+BOOST_LOG_GLOBAL_LOGGER(my_logger, boost::log::sources::severity_logger_mt<VegaStrikeLogging::vega_log_level>)
+
 BOOST_LOG_ATTRIBUTE_KEYWORD(severity, "Severity", vega_log_level)
 
-typedef boost::log::sinks::synchronous_sink<boost::log::sinks::text_ostream_backend> ConsoleLogSink;
-typedef boost::log::sinks::synchronous_sink<boost::log::sinks::text_file_backend> FileLogSink;
+typedef boost::log::sinks::text_ostream_backend ConsoleLogBackEnd;
+typedef boost::log::sinks::text_file_backend FileLogBackEnd;
+typedef boost::log::sinks::asynchronous_sink<ConsoleLogBackEnd> ConsoleLogSink;
+typedef boost::log::sinks::asynchronous_sink<FileLogBackEnd>    FileLogSink;
 
-#define VS_LOG(log_level, log_message)                                                                              \
-    do {                                                                                                            \
-        VegaStrikeLogging::vega_logger()->Log(VegaStrikeLogging::vega_log_level::log_level, (log_message));         \
+#define VS_LOG(log_level, log_message)                                                                                                          \
+    do {                                                                                                                                        \
+        VegaStrikeLogging::VegaStrikeLogger::instance().Log(VegaStrikeLogging::vega_log_level::log_level, (log_message));                       \
     } while (false)
-#define VS_LOG_AND_FLUSH(log_level, log_message)                                                                    \
-    do {                                                                                                            \
-        VegaStrikeLogging::vega_logger()->LogAndFlush(VegaStrikeLogging::vega_log_level::log_level, (log_message)); \
+#define VS_LOG_AND_FLUSH(log_level, log_message)                                                                                                \
+    do {                                                                                                                                        \
+        VegaStrikeLogging::VegaStrikeLogger::instance().LogAndFlush(VegaStrikeLogging::vega_log_level::log_level, (log_message));               \
+    } while (false)
+#define VS_LOG_FLUSH_EXIT(log_level, log_message, exit_code)                                                                                    \
+    do {                                                                                                                                        \
+        VegaStrikeLogging::VegaStrikeLogger::instance().LogFlushExit(VegaStrikeLogging::vega_log_level::log_level, (log_message), (exit_code)); \
     } while (false)
 
-class VegaStrikeLogger {
+struct VegaStrikeLogger {
```

Comment:
we should keep this as a `class`; the only real difference is default scoping.
`struct` -> default public
`class` -> default private

Reply:
OK. Yeah, I'll change it back.

Reply:
Done ✔️